src/XR/features/WebXRLightEstimation.ts

```diff
@@ -0,0 +1,366 @@
+     * Since creating and processing the cube map is potentially expensive, especially if mip maps are needed, you can listen to the onReflectionCubeMapUpdatedObservable to determine
+     * when it has been updated.
+     */
+    public get reflectionCubeMapTexture(): Nullable<BaseTexture> {
+        return this._reflectionCubeMap;
+    }
+
+    /**
+     * The most recent light estimate.  Available starting on the first frame where the device provides a light probe.
+     */
+    public get xrLightingEstimate(): Nullable<IWebXRLightEstimation> {
+        if (this._xrLightEstimate) {
+            return {
+                lightColor: this._lightColor,
+                lightDirection: this._lightDirection,
+                lightIntensity: this._intensity
+            };
+
+        }
+        return this._xrLightEstimate;
+    }
+
+    private _getCanvasContext(): WebGLRenderingContext | WebGL2RenderingContext {
+        if (this._canvasContext === null) {
+            this._canvasContext = this._xrSessionManager.scene.getEngine()._gl;
+        }
+        return this._canvasContext;
+    }
+
+    private _getXRGLBinding(): XRWebGLBinding {
+        if (this._xrWebGLBinding === null) {
+            let context = this._getCanvasContext();
+            this._xrWebGLBinding = new XRWebGLBinding(this._xrSessionManager.session, context);
+        }
+        return this._xrWebGLBinding;
+    }
+
+    /**
+     * Event Listener to for "reflectionchange" events.
+     */
+    private _updateReflectionCubeMap = (): void => {
+        if (!this._xrLightProbe) {
+            return;
+        }
+        // check poll time, do not update if it has not been long enough
+        if (this.options.cubeMapPollInterval) {
+            const now = Date.now();
+            if (now - this._cubeMapPollTime < this.options.cubeMapPollInterval) {
+                return;
+            }
+            this._cubeMapPollTime = now;
+        }
+        const lp = this._getXRGLBinding().getReflectionCubeMap(this._xrLightProbe);
+        if (lp && this._reflectionCubeMap) {
+            if (this._reflectionCubeMap._texture) {
+                this._reflectionCubeMap._texture.dispose();
+            }
+            const internalTexture = new InternalTexture(this._xrSessionManager.scene.getEngine(), InternalTextureSource.Unknown);
+            internalTexture.isCube = true;
+            internalTexture.invertY = false;
+            internalTexture._useSRGBBuffer = this.options.reflectionFormat === 'srgba8';
+            internalTexture.format = Constants.TEXTUREFORMAT_RGBA;
+            internalTexture.generateMipMaps = false;
+            internalTexture.type = this.options.reflectionFormat !== 'srgba8' ? Constants.TEXTURETYPE_HALF_FLOAT : Constants.TEXTURETYPE_UNSIGNED_BYTE;
+            internalTexture.samplingMode = Constants.TEXTURE_NEAREST_SAMPLINGMODE;

```

Comment:
I am not sure we should use NEAREST here, actually an env map should have mip maps and filtering turned on

Reply:
this is a 16x16 texture generated by webxr itself. i didn't see the nead for mipmaps, but i can enable it.
Regarding sampling - i can remove it, of course.